### PR TITLE
Okina-CMake: CUDA, OCCA, RAJA + MPI [okina-cmake]

### PR DESCRIPTION
**Pull request by camierjs**

### CODE REVIEW

**Comment by tzanio** [comment]
TODO: Update `INSTALL`

File: config/XSDKDefaults.cmake [hunk 1]
```diff
@@ -96,7 +96,7 @@ IF ("${USE_XSDK_DEFAULTS_DEFAULT}" STREQUAL "")
 ENDIF()
 SET(USE_XSDK_DEFAULTS  ${USE_XSDK_DEFAULTS_DEFAULT}  CACHE  BOOL
   "Use XSDK defaults and behavior.")
-PRINT_VAR(USE_XSDK_DEFAULTS)
+#PRINT_VAR(USE_XSDK_DEFAULTS)
 
 SET_DEFAULT(XSDK_ENABLE_C  TRUE)
```
[the remark this answers is not in the document]
Comment (camierjs) [reply config/XSDKDefaults.cmake]: I don't know the reason of printing this variable, if it is for debug purpose, we should get rid of it

File: CMakeLists.txt [hunk 2]
```diff
@@ -86,6 +86,13 @@ include("${CMAKE_CURRENT_SOURCE_DIR}/config/XSDKDefaults.cmake")
 
 # Enable languages.
 enable_language(CXX)
+if (MFEM_USE_CUDA)
+   # MFEM_USE_CUDA requires CMake 3.8 or newer (for direct CUDA support)
+   cmake_minimum_required(VERSION 3.8 FATAL_ERROR)
+   set(CUDA_PROPAGATE_HOST_FLAGS ON)
```
Comment (v-dobrev) [inline CMakeLists.txt]: This option does not seem to work - looks like it is only used by the old `CUDA` module (`FindCUDA.cmake`).

I'll remove it unless there are any objections.

Comment (camierjs) [reply CMakeLists.txt]: You're right, that may not be useful anymore.

**Comment by camierjs** [comment]
Thanks for the last changes, @v-dobrev

**Comment by tzanio** [comment]
This depends on  #868, so we should merge them in next/master together.

**Comment by v-dobrev** [comment]
Actually, I added a temporary fix in this branch so it does not need other branches to work:
https://github.com/mfem/mfem/blob/aa027c2b8d26c0a4c72c4a9df0fecdb72939f823/CMakeLists.txt#L387-L390

**Pull request merged**